src/pages/build-apps/examples/heystack.md

```diff
@@ -0,0 +1,482 @@
+specification, Clarity has the following types:
+
+- `(tuple (key-name-0 key-type 0) (key-name-1 key-type-1) ...)` - a typed tuple with named fields
+- `(list max-len entry-type)` - a list of maximum length `max-len`, with entries of type `entry-type`
+- `(response ok-type err-type)` - object used by public functions to commit their state changes or abort
+- `(optional some-type)` - an option type for objects that can be either `(some-value)` or `none`
+- `(buff max-len)` - byte buffer of maximum length
+- `principal` - object representing a principal address (contract or standard)
+- `bool` - boolean value (`true` or `false`)
+- `int` - signed 128-bit integer
+- `uint` - unsigned 128-bit integer
+
+To support these types in Javascript, [`@stacks/transactions`][] exports the following helpers:
+
+```ts
+// construct boolean clarity values
+const t = trueCV();
+const f = falseCV();
+
+// construct optional clarity values
+const nothing = noneCV();
+const something = someCV(t);
+
+// construct a buffer clarity value from an existing Buffer
+const buffer = Buffer.from('foo');
+const bufCV = bufferCV(buffer);
+
+// construct signed and unsigned integer clarity values
+const i = intCV(-10);
+const u = uintCV(10);
+
+// construct principal clarity values
+const address = 'SP2JXKMSH007NPYAQHKJPQMAQYAD90NQGTVJVQ02B';
+const contractName = 'contract-name';
+const spCV = standardPrincipalCV(address);
+const cpCV = contractPrincipalCV(address, contractName);
+
+// construct response clarity values
+const errCV = responseErrorCV(trueCV());
+const okCV = responseOkCV(falseCV());
+
+// construct tuple clarity values
+const tupCV = tupleCV({
+  a: intCV(1),
+  b: trueCV(),
+  c: falseCV(),
+});
+
+// construct list clarity values
+const l = listCV([trueCV(), falseCV()]);
+```
+
+You should use these helpers when calling Clarity contracts with Javascript to avoid failed contract calls due to bad
+typing.
+
+## Reading BNS names
+
+An important feature of Stacks is the [Blockchain Naming System][] (BNS). BNS allows users to register an human-readable
+identity to their account, that can act as both a username and a web address.
+
+Names registered to a user can be read from a Stacks API endpoint, as demonstrated in [`src/store/names.ts`][].
+
+-> Due to ecosystem limitations, it's currently uncommon for BNS names to be registered on any testnet. For the purpose
+of demonstration, Heystack looks for BNS names against the user's mainnet wallet address.
+
+```ts
+export const namesAtom = atomFamily((address: string) =>
+  atom(get => {
+    const network = get(mainnetNetworkAtom);
+    if (!network) return null;
+
+    const local = getLocalNames(network.coreApiUrl, address);
+
+    if (local) {
+      const [names, timestamp] = local;
+      const now = Date.now();
+      const isStale = now - timestamp > STALE_TIME;
+      if (!isStale) return names;
+    }
+
+    try {
+      const names = useNamesByAddress(network.coreApiUrl, address);
```

Comment:
This hook implementation was incorrect, so it needs to be updated with current main. Recopy in `namesAtom` code.